src/main/java/ch/njol/skript/expressions/ExprBeaconValues.java

```diff
@@ -0,0 +1,199 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.util.Getter;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.block.Beacon;
+import org.bukkit.event.Event;
+import org.bukkit.potion.PotionEffectType;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnknownNullability;
+
+@Name("Beacon Values")
+@Description({
+	"Values of a Beacon",
+	"Secondary effect can only be set to regeneration.",
+	"Secondary effect can only be set when beacon tier is maxxed.",
+	"You can only change the range on Paper.",
+	"You can only set primary and secondary effects to null (clear) on Paper."
+})
+@Examples({
+	"broadcast tier of (beacon from {_block})",
+	"set primary effect of (beacon from {_block}) to haste",
+	"add 1 to range of (beacon from {_block})"
+})
+@Since("INSERT VERSION")
+public class ExprBeaconValues extends PropertyExpression<Beacon, Object> {
+
+	private static boolean PAPER_LOADED = false;
```

Comment:
```suggestion
	private static boolean SUPPORTS_BEACON_TIERS = false;
```

Reply:
That is a more direct name,  I feel PAPER_LOADED is fine, and it's also needed to check if you can clear the effects.
I'm keeping as is.

Reply:
The name should describe exactly what it ensures. It doesn't ensure paper is loaded, it ensures that the getTier method exists, hence SUPPORTS_BEACON_TIERS is the better name. If you're using it to check for clearing the effects, then you should check for that method as well, or use a check that covers both.